serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordController.java

```diff
@@ -42,159 +52,160 @@
 
     private static final Logger log = LoggerFactory.getLogger(ServiceRecordController.class);
     private final ServiceRecordFactory serviceRecordFactory;
+    private final ProcessService processService;
+    private final AuthenticationService authenticationService;
     private EntityService entityService;
     private PayloadSigner payloadSigner;
+    private SRRequestScope requestScope;
 
     /**
-     * @param serviceRecordFactory for creation of the identifiers respective service record
-     * @param entityService        needed to lookup and retrieve organization or citizen information using an identifier number
+     * @param serviceRecordFactory  for creation of the identifiers respective service record
+     * @param entityService         needed to lookup and retrieve organization or citizen information using an identifier number
+     * @param processService
+     * @param authenticationService
+     * @param requestScope
      */
-    @Autowired
     public ServiceRecordController(ServiceRecordFactory serviceRecordFactory,
                                    EntityService entityService,
-                                   PayloadSigner payloadSigner) {
+                                   PayloadSigner payloadSigner,
+                                   ProcessService processService,
+                                   AuthenticationService authenticationService,
+                                   SRRequestScope requestScope) {
         this.entityService = entityService;
         this.serviceRecordFactory = serviceRecordFactory;
         this.payloadSigner = payloadSigner;
+        this.processService = processService;
+        this.authenticationService = authenticationService;
+        this.requestScope = requestScope;
     }
 
     @InitBinder
     protected void initBinders(WebDataBinder binder) {
         binder.registerCustomEditor(Notification.class, new NotificationEditor());
     }
 
+    /**
+     * Used to retrieve information needed to send a message within the provided process
+     * to an entity with the provided identifier.
+     *
+     * @param identifier        specifies the target entity.
+     * @param processIdentifier specifies the target process.
+     * @param auth              provides the authentication object.
+     * @param request           is the servlet request.
+     * @return JSON object with information needed to send a message.
+     */
+    @GetMapping(value = "/identifier/{identifier}/process/{processIdentifier}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public ResponseEntity entity(@PathVariable("identifier") String identifier,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: no.difi.meldingsutveksling.serviceregistry.exceptions.SecurityLevelNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.KRRClientException, no.difi.meldingsutveksling.serviceregistry.CertificateNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.DsfLookupException, no.difi.meldingsutveksling.serviceregistry.service.brreg.BrregNotFoundException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)